### PR TITLE
Notify user on plugin unload failures

### PR DESCRIPTION
**Pull request by Knightly1**
- Plugins that don't unload will now not be able to load again until that is resolved
- /unload will prompt the user to kill the session or retry rather than let them unload uninformed
- User can override this behavior with /unload force
- Added /plugin list failed and /plugin list dlls for more information on plugins
- Switches /plugin parameters to getarg (resolves issue with /plugin MQNoAutomaticStuff)

### CODE REVIEW

**Review by dannuic — APPROVED** [review]
Mostly minor feedback, will approve and leave it optional to you if you want to change anything.

File: include/mq/base/Config.h [hunk 1]
```diff
@@ -126,6 +126,12 @@ inline std::vector<std::string> GetPrivateProfileKeys(const std::string& section
 	return results;
 }
 
+inline bool GetPrivateProfileKeyExists(const std::string& section, const std::string& key, const std::string& iniFileName)
```
[the remark this answers is not in the document]
Comment (Knightly1) [reply include/mq/base/Config.h:129]: Yeah, this was trying to conform to WinAPI dialect.

File: src/main/MQ2Main.cpp [hunk 2]
```diff
@@ -788,6 +788,22 @@ void MQ2Shutdown()
 	ShutdownMQ2Commands();
 	ShutdownAnonymizer();
 	ShutdownMQ2Plugins();
+	if(!UnloadFailedPlugins() && !gbForceUnload)
```
[the remark this answers is not in the document]
Comment (Knightly1) [reply src/main/MQ2Main.cpp:791]: Agreed, I'll move it.

File: include/mq/base/Config.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (Knightly1) [reply include/mq/base/Config.h]: Updated to PrivateProfileKeyExists

File: src/main/MQ2Main.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (Knightly1) [reply src/main/MQ2Main.cpp]: Added ShutdownFailedPlugins()

**Pull request merged**